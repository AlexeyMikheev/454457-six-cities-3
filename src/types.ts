import {OfferType} from "./consts";

export interface Location {
    latitude: number;
    longitude: number;
    zoom: number;
    center: number[];
}

export interface City {
    name: string;
    location: Location;
    center: number[];
    zoom: number;
}

export interface Owner {
    id: number;
    name: string;
    avatar: string;
    isTrust: boolean;
}

export interface Offer {
    owner: Owner;
    location: Location;
    city: City;
    id: number;
    isPremium: boolean;
    isMarked: boolean;
    cost: number;
    rating: number;
    name: string;
    description: string;
    roomsCount: number;
    membersCount: number;
    type: OfferType;
    image: string;
    images: string[];
    features: string[];
    cityName: string;
    center: number[];
    zoom: number;
}

export interface UserInfo {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface Review{
    id: number;
    comment: string;
    rating: number;
    date: number;
    user: Owner;
  }

export type HoveredChangeArgType = number | null;
export type SendCommentArgType = {
    rating: string;
    comment: string;
}
export type FormValueArgsType = {
    target: {
        name: string;
        value: string;
    };
}

export type SubmitArgsType = {
    login: string;
    password: string;
}

export interface FormState{
    [key: string]: string;
 }

export interface BooleanState{
    isToggled: boolean;
 }

export type OfferHoveredChangeFunction = (offerId: HoveredChangeArgType) => void | null;
export type OfferChangeFunction = (offerId: number) => void;
export type CityChangeFunction = (cityName: string) => void;
export type SendCommentFunction = (offerId: number, args: SendCommentArgType) => void;
export type FormValueChangeFunction = (args: FormValueArgsType) => void;
export type FormSubmitFunction = (args: SubmitArgsType) => void;

export type EmptyFunction = () => void;

export type MapSetting = {
    zoomControl: boolean;
    marker: boolean;
}
