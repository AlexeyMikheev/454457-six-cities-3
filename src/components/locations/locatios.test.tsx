import * as React from "react";
import {mount} from "enzyme";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import Locations from "./locations";
import NameSpace from "../../reducer/name-space.js";
import {City, Location} from "../../types";

const mockStore = configureStore([]);

const location: Location = {
  latitude: 4.85309666406198,
  longitude: 52.3909553943508,
  zoom: 10,
  center: [4.85309666406198, 52.3909553943508]
};

const cityParis: City = {
  name: `Paris`,
  location,
  center: location.center,
  zoom: location.zoom
};

const mocksCities: City[] = [
  cityParis,
  {
    name: `Cologne`,
    location,
    center: location.center,
    zoom: location.zoom
  },
  {
    name: `Brussels`,
    location,
    center: location.center,
    zoom: location.zoom
  },
  {
    name: `Amsterdam`,
    location,
    center: location.center,
    zoom: location.zoom
  },
  {
    name: `Hamburg`,
    location,
    center: location.center,
    zoom: location.zoom
  },
  {
    name: `Dusseldorf`,
    location,
    center: location.center,
    zoom: location.zoom
  }
];

it(`Locations snapshot`, () => {

  const store = mockStore({[NameSpace.DATA]: {
    offers: [],
    currentOffers: [],
    currentOffer: null,
    cities: mocksCities,
    reviews: [],
    currentCity: cityParis,
    nearOffers: []
  }});

  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const locationsComponent = mount(<Provider store={store}><Locations /></Provider>, {attachTo: div});

  expect(locationsComponent.getDOMNode()).toMatchSnapshot();
});
