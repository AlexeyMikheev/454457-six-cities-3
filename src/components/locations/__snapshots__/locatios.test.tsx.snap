// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations snapshot 1`] = `
Array [
  null,
]
`;
